File: TUF/Numbers/FactorialOfNumber.ts
function findFactorial(num: number): number {
    if (num == 0) {
        return 1;
    } else {
        return num * findFactorial(num - 1);
    }
}

let num: number = 5;
console.log(`Factorial of ${num} is: ${findFactorial(num)}`);

export { };

File: TUF/Numbers/FactorialOfNumber.java
import java.util.Scanner;

public class FactorialOfNumber {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter a number to find the factorial: ");
        int num = input.nextInt();
        input.close();
        System.out.printf("Factorial of %d is: %d\n", num, findFactorial(num));
    }

    private static int findFactorial(int num) {
        if (num == 0) {
            return 1;
        } else {
            return num * findFactorial(num - 1);
        }
    }
}
